src/nixfeature.cc

```diff
@@ -12,18 +12,20 @@ namespace nixfeature {
 
     mxArray *describe(const nix::Feature &feat)
     {
-        struct_builder sb({ 1 }, { "id" });
+        nix::LinkType link_type = feat.linkType();
+        uint8_t ltype;
+
+        switch (link_type) {
+        case nix::LinkType::Tagged: ltype = 0; break;
+        case nix::LinkType::Untagged: ltype = 1; break;
+        case nix::LinkType::Indexed: ltype = 2; break;
+        default: throw std::invalid_argument("unkown link type");
+        }
+
+        struct_builder sb({ 1 }, { "id", "linkType" });
         sb.set(feat.id());
+        sb.set(ltype);
```

Comment:
I'd propose to keep this in mkarray where all other conversions are. I guess we just need to add the method like

mxArray* make_mx_array(const nix::LinkType &ltype);

that just does exactly what is already written. Then one could just set the structure builder element like

sb.set(feat.linkType());